From 5801be0ed4a00a553f5a8dabbff4713b0ed4154b Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Sun, 14 Jan 2018 16:49:36 +0100
Subject: [PATCH 1/1] efi_loader: text output for device path end node

Without the patch a device path consisting only of an end node is
displayed as '/UNKNOWN(007f,00ff)'. It should be displayed as '/'.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 lib/efi_loader/efi_device_path_to_text.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lib/efi_loader/efi_device_path_to_text.c b/lib/efi_loader/efi_device_path_to_text.c
index 40eae730ed..a79e60a4ee 100644
--- a/lib/efi_loader/efi_device_path_to_text.c
+++ b/lib/efi_loader/efi_device_path_to_text.c
@@ -245,6 +245,8 @@ static char *efi_convert_single_device_node_to_text(
 	case DEVICE_PATH_TYPE_MEDIA_DEVICE:
 		str = dp_media(str, dp);
 		break;
+	case DEVICE_PATH_TYPE_END:
+		break;
 	default:
 		str = dp_unknown(str, dp);
 	}
-- 
2.14.2

